src/Control/Optics/Linear/Internal.hs

```diff
@@ -139,3 +152,10 @@ withIso (Optical l) f = f fro to
 withPrism :: Optic_ (Market a b) a b s t -> ((b #-> t) -> (s #-> Either t a) -> r) -> r
 withPrism (Optical l) f = f b m
   where Market b m = l (Market id Right)
+
+withLens :: Optic_ (Linear.Kleisli (Compose ((,) a) (FUN 'One b))) a b s t -> s #-> (a, b #-> t)
```

Comment:
Given that this is just migration and that I'm going to take a pass through optics, this, although complicated, is fine right now. When I take a pass I'm going to comment that `withLens` is (kind of) the dual of `lens` and the `(Optic_ ...)` is basically `(a #-> (a, b -> b)) -> (s #-> (s, t -> t))`. 

One question: could we change `FUN 'One b` to `(->) b` which is much more readable to me?

Reply:
> `withLens` is (kind of) the dual of `lens`

The better term is _inverse_

> could we change `FUN 'One b to (->) b` which is much more readable to me?

It would not work. `(->)` is `FUN 'Many`. There are many ways to make this simpler, but I don't think that they are much needed. The type after `Optic_` is not meant to be read. See the `lens` library for how to document such primitives.